patches/systemd/0002-Add-support-for-systemd-boot-for-Managarm.patch

```diff
@@ -0,0 +1,651 @@
+          * the ones we want to keep */
+         r = capability_bounding_set_drop(keep_capabilities, true);
+@@ -365,6 +366,7 @@ int drop_privileges(uid_t uid, gid_t gid, uint64_t keep_capabilities) {
+                 if (cap_set_proc(d) < 0)
+                         return log_error_errno(errno, "Failed to increase capabilities: %m");
+         }
++#endif
+ 
+         return 0;
+ }
+diff --git a/src/basic/process-util.c b/src/basic/process-util.c
+index 6a1e496..c9f30bb 100644
+--- a/src/basic/process-util.c
++++ b/src/basic/process-util.c
+@@ -1621,6 +1621,8 @@ int safe_fork_full(
+                 }
+         }
+ 
++// TODO: HACK!
++#ifndef __managarm__
+         if (FLAGS_SET(flags, FORK_NEW_MOUNTNS | FORK_MOUNTNS_SLAVE)) {
+                 /* Optionally, make sure we never propagate mounts to the host. */
+                 if (mount(NULL, "/", NULL, MS_SLAVE | MS_REC, NULL) < 0) {
+@@ -1628,6 +1630,7 @@ int safe_fork_full(
+                         _exit(EXIT_FAILURE);
+                 }
+         }
++#endif
+ 
+         if (FLAGS_SET(flags, FORK_PRIVATE_TMP)) {
+                 assert(FLAGS_SET(flags, FORK_NEW_MOUNTNS));
+@@ -1973,8 +1976,8 @@ _noreturn_ void freeze(void) {
+         }
+ 
+         /* waitid() failed with an unexpected error, things are really borked. Freeze now! */
+-        for (;;)
+-                pause();
++        for (;;);
++                // pause();
+ }
+ 
+ int get_process_threads(pid_t pid) {
+diff --git a/src/basic/raw-clone.h b/src/basic/raw-clone.h
+index c5bfc10..c03adcc 100644
+--- a/src/basic/raw-clone.h
++++ b/src/basic/raw-clone.h
+@@ -78,11 +78,9 @@ static inline pid_t raw_clone(unsigned long flags) {
+         }
+ #elif defined(__managarm__)
+         int helout = open("/dev/helout", O_RDWR);
+-        dprintf(helout, "systemd: clone unsupported\n");
++        dprintf(helout, "systemd: clone unsupported, falling back to fork\n");
+         close(helout);
+-
+-        errno = ENOSYS;
+-        ret = -1;
++        ret = fork();
+ #else
+         ret = (pid_t) syscall(__NR_clone, flags, NULL);
+ #endif
+diff --git a/src/basic/stdio-util.h b/src/basic/stdio-util.h
+index 0a2239d..2e55e94 100644
+--- a/src/basic/stdio-util.h
++++ b/src/basic/stdio-util.h
+@@ -7,6 +7,7 @@
+ #include <sys/types.h>
+ 
+ #include "macro.h"
++#include "parse-printf-format.h"
+ 
+ _printf_(3, 4)
+ static inline char* snprintf_ok(char *buf, size_t len, const char *format, ...) {
+diff --git a/src/basic/time-util.c b/src/basic/time-util.c
+index 5201d5a..a4e5680 100644
+--- a/src/basic/time-util.c
++++ b/src/basic/time-util.c
+@@ -1704,7 +1704,7 @@ int time_change_fd(void) {
+ 
+         /* We only care for the cancellation event, hence we set the timeout to the latest possible value. */
+         static const struct itimerspec its = {
+-                .it_value.tv_sec = TIME_T_MAX,
++                .it_value.tv_sec = INT32_MAX, // HACK!
+         };
```

Comment:
Why is this needed? If this causes an overflow in mlibc or managarm-system, we should fix the overflow.

Reply:
This is no longer needed and has been dropped.